src/Utils.h

```diff
@@ -0,0 +1,217 @@
+#ifndef Utils_H
+#define Utils_H
+#define PATH_MAX_LENGTH 255
+
+#include "Arduino.h"
+#include "Arduino_UnifiedStorage.h"
+#include "Arduino_POSIXStorage.h"
+#include <iostream>
+
+#if !defined(HAS_SERIAL) && defined(HAS_RS485)
+#include <ArduinoRS485.h>
+
+
+[[gnu::unused]] static void beginRS485(const int baudrate){
+    const auto bitduration { 1.f / baudrate };
+    const auto wordlen { 9.6f }; // OR 10.0f depending on the channel configuration
+    const auto preDelayBR { bitduration * wordlen * 3.5f * 1e6 };
+    const auto postDelayBR { bitduration * wordlen * 3.5f * 1e6 };
+
+    RS485.begin(baudrate);
+    RS485.setDelays(preDelayBR, postDelayBR);
+    RS485.noReceive();
+}
+
+[[gnu::unused]] static void debugPrintRS485(String s){
+    static bool rs485Initialized = false;
+    if (!rs485Initialized) {
+      beginRS485(115200);
+      rs485Initialized = true;
+    }
+    RS485.beginTransmission();
+    RS485.write(s.c_str(), strlen(s.c_str()));
+    RS485.write('\n');
+    RS485.endTransmission();
+}
+
+#endif
+
+
+[[gnu::unused]] static String prettyPrintFileSystemType(FileSystems f){
+    if(f == 0) return "FAT";
+    else if(f == 1) return "LitlleFS";
+    else return "";
+}
+
+
+[[gnu::unused]] static const char* createPartitionName(int number) {
+    if (number < 1 || number > 26) {
+        // Handle out-of-range numbers or errors as needed
+        return nullptr;
+    }
+
+    char partitionName[13]; // "partition_" + letter + '\0'
+    snprintf(partitionName, sizeof(partitionName), "%c", 'a' + number - 1);
+
+    // Dynamically allocate memory for the string and copy the generated value
+    char* dynamicName = new char[strlen(partitionName) + 1];
+    strcpy(dynamicName, partitionName);
+
+    return dynamicName;
+}
+
+
+[[gnu::unused]] static bool copyFolder(const char* source, const char* destination) {
```

Comment:
copyFolder recursivelly copies all folder inside a folder, I made this method here so that you wouldn't need to do recusion of a member method which in turn would have had created some other problems, at that point, It made more sense to add it here as a static here, but I could have also made it a static member function. I would leave it here for now, to not do any last minute changes that might break stuff, but I'll move it in the next release/fix etc.